libsolidity/ast/Types.cpp

```diff
@@ -2140,12 +2140,26 @@ FunctionType const* ContractType::newExpressionType() const
 	return m_constructorType;
 }
 
-std::vector<std::tuple<VariableDeclaration const*, u256, unsigned>> ContractType::stateVariables() const
+std::vector<std::tuple<VariableDeclaration const*, u256, unsigned>> ContractType::stateVariables(std::optional<DataLocation> _location) const
 {
+	auto location = [&]() -> std::optional<VariableDeclaration::Location> {
```

Comment:
Yeah, I agree that this is misleading until the last moment (and if you catch it) when you see the `()`.
I will add the explicit type, I used `auto` mainly because I have to have the explicit `-> std::optional<VariableDeclaration::Location>` otherwise, the `std::nullopt` is deduced as the return type and conflicts with the actual desired return type.

Reply:
I mean I'd just drop the lambda :)

Bit anyway, the way it is now is much better.

Reply:
Yeah, you are right, after re-writing it, it is not adding anything. I've dropped it.